commands/apps_dev.go

```diff
@@ -0,0 +1,505 @@
+package commands
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"sync"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/digitalocean/doctl"
+	"github.com/digitalocean/doctl/commands/charm"
+	"github.com/digitalocean/doctl/commands/charm/confirm"
+	"github.com/digitalocean/doctl/commands/charm/list"
+	"github.com/digitalocean/doctl/commands/charm/pager"
+	"github.com/digitalocean/doctl/commands/charm/template"
+	"github.com/digitalocean/doctl/commands/charm/text"
+	"github.com/digitalocean/doctl/commands/charm/textbox"
+	"github.com/digitalocean/doctl/internal/apps/builder"
+	"github.com/digitalocean/doctl/internal/apps/config"
+	"github.com/digitalocean/doctl/internal/apps/workspace"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/pkg/jsonmessage"
+
+	"github.com/digitalocean/godo"
+	"github.com/spf13/cobra"
+)
+
+const (
+	// AppsDevDefaultEnvFile is the default env file path.
+	AppsDevDefaultEnvFile = ".env"
+)
+
+// AppsDev creates the apps dev command subtree.
+func AppsDev() *Command {
+	cmd := &Command{
+		Command: &cobra.Command{
+			Use:     "dev",
+			Aliases: []string{},
+			Short:   "[BETA] Display commands for working with app platform local development.",
+			Long:    `[BETA] Display commands for working with app platform local development.`,
+			Hidden:  true,
+		},
+	}
+
+	cmd.AddCommand(AppsDevConfig())
+
+	build := CmdBuilder(
+		cmd,
+		RunAppsDevBuild,
+		"build [component name]",
+		"Build an app component",
+		heredoc.Doc(`
+			[BETA] Build an app component locally.
+			
+			  The component name must be specified as an argument if running non-interactively.`,
+		),
+		Writer,
+		aliasOpt("b"),
+	)
+	build.DisableFlagsInUseLine = true
+
+	AddStringFlag(
+		build, doctl.ArgAppSpec,
+		"", "",
+		`Path to an app spec in JSON or YAML format. Set to "-" to read from stdin.`,
```

Comment:
```suggestion
		`Path to an app spec in JSON or YAML format.`,
```

Reply:
I updated to support reading from `stdin` in https://github.com/digitalocean/doctl/pull/1237/commits/0f8d843603fcff91c1fb60713448b7c4872d5ef1 ; that should work as expected now